solver/src/settlement_submission/flashbots_api.rs

```diff
@@ -30,7 +30,26 @@ impl FlashbotsApi {
         let status = response.status();
         let body = response.text().await?;
         ensure!(status.is_success(), "status {}: {:?}", status, body);
-        Ok(body)
+
+        match serde_json::from_str::<jsonrpc_core::Output>(&body) {
+            Ok(body) => {
+                let bundle_id = if let jsonrpc_core::Output::Success(ref x) = body {
+                    x.result.as_str().unwrap_or_default()
+                } else {
+                    Default::default()
```

Comment:
shouldn't the failure cases here also lead to a high level failure of the submission instead of assuming an empty string?

Reply:
Fixed that ugly unwraping.